kafka-client/src/main/java/dev/responsive/kafka/internal/stores/PartitionedOperations.java

```diff
@@ -160,10 +158,10 @@ public static PartitionedOperations create(
       storeRegistry.registerStore(registration);
 
       final boolean migrationMode = ConfigUtils.responsiveMode(config) == ResponsiveMode.MIGRATE;
-      long startingTimestamp = -1;
-      final Optional<Duration> ttl = params.timeToLive();
-      if (migrationMode && ttl.isPresent()) {
-        startingTimestamp = Instant.now().minus(ttl.get()).toEpochMilli();
+      long startTimeMs = -1;
+      if (migrationMode && params.ttlProvider().hasTtl()) {
+        // TODO(sophie): figure out how to account for row-level ttl in migration mode
+        startTimeMs = System.currentTimeMillis() - params.ttlProvider().defaultTtl().toMillis();
```

Comment:
let's create a follow-up ticket, but I think what we can do is just have them pass in the "start time" to the tool, and then just apply individual TTls as necessary